include/swift/AST/DiagnosticsSema.def

```diff
@@ -7848,6 +7848,10 @@ ERROR(noncopyable_objc_enum, none,
       "noncopyable enums cannot be marked '@objc'", ())
 ERROR(moveOnly_not_allowed_here,none,
       "'@_moveOnly' attribute is only valid on structs or enums", ())
+WARNING(consume_expression_needed_for_cast,none,
```

Comment:
Do we mean to keep this a warning forever, or is this meant to be an error at some point?

Reply:
It would be better if this started out as an error, since this is the first time anyone can implicitly cast a noncopyable type.